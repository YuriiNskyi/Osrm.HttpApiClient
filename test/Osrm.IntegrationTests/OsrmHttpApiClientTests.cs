using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Osrm.Examples;
using Osrm.HttpApiClient;

namespace Osrm.IntegrationTests
{
    [UseApprovalSubdirectory("Approvals")]
    [UseReporter(typeof(VisualStudioReporter))]
    public class OsrmHttpApiClientTests
    {
        private HttpClient _httpClient;
        private OsrmHttpApiClient _osrmClient;

        [SetUp]
        public void Setup()
        {
            _httpClient = new HttpClient
            {
                BaseAddress = new Uri("http://router.project-osrm.org"),
            };

            _osrmClient = new OsrmHttpApiClient(_httpClient);
        }

        [TearDown]
        public void Teardown()
        {
            _osrmClient = null;
            _httpClient.Dispose();
        }

        [Test]
        public async Task Full_GeoJson_route_query()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Full_GeoJson_route_query);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Full_Polyline_route_query()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Full_Polyline_route_query);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_with_three_coordinates()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_with_three_coordinates);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_excluding_the_usage_of_motorways()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_excluding_the_usage_of_motorways);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Query_on_Berlin_using_polyline()
        {
            var response = await _osrmClient.GetRouteAsync(ProjectOsrmOrgDocs.Query_on_Berlin_using_polyline);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Querying_nearest_three_snapped_locations()
        {
            var response = await _osrmClient.GetNearestAsync(ProjectOsrmOrgDocs.Querying_nearest_three_snapped_locations);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_1x3_duration_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_1x3_duration_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task An_asymmetric_3x2_duration_matrix_with_from_the_polyline_encoded_locations()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.An_asymmetric_3x2_duration_matrix_with_from_the_polyline_encoded_locations);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix_for_CH()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix_for_CH);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task A_3x3_duration_matrix_and_a_3x3_distance_matrix_for_CH()
        {
            var response = await _osrmClient.GetTableAsync(ProjectOsrmOrgDocs.A_3x3_duration_matrix_and_a_3x3_distance_matrix_for_CH);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Round_trip_in_Berlin_with_three_stops()
        {
            var response = await _osrmClient.GetTripAsync(ProjectOsrmOrgDocs.Round_trip_in_Berlin_with_three_stops);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }

        [Test]
        public async Task Round_trip_in_Berlin_with_four_stops_starting_at_the_first_stop_ending_at_the_last()
        {
            var response = await _osrmClient.GetTripAsync(ProjectOsrmOrgDocs.Round_trip_in_Berlin_with_four_stops_starting_at_the_first_stop_ending_at_the_last);

            var json = JsonSerializer.Serialize(response, OsrmHttpApiClient.DefaultJsonSerializerOptions);

            Approvals.VerifyJson(json);
        }
    }
}