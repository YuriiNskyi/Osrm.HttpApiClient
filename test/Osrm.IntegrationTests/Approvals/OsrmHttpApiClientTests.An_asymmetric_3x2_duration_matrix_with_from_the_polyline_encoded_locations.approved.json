{
    "durations": [
        [
            1001,
            439.7
        ],
        [
            842.6,
            790.6
        ],
        [
            681.6,
            381.8
        ]
    ],
    "distances": [],
    "sources": [
        {
            "nodes": [],
            "name": "Conrad-Blenkle-Stra\u00DFe",
            "location": [
                13.445158,
                52.532489
            ],
            "distance": 4.232946,
            "hint": "Kdz0kp1CSoASAAAABgAAACcAAAAAAAAAtjukQVa9xUAbZC1CAAAAABIAAAAGAAAAJwAAAAAAAADXFgEAJijNAAmVIQNaKM0AHpUhAwEALwl2gVsJ"
        },
        {
            "nodes": [],
            "name": "Columbiadamm",
            "location": [
                13.398532,
                52.48384
            ],
            "distance": 24.526558,
            "hint": "Pp7XgRITApMZAAAADwAAAPoAAAAmAAAAOujgQQvMg0EYJoBDZiwrQhkAAAAPAAAA-gAAACYAAADXFgEABHLMAADXIAPuccwAJNYgAwQA3wV2gVsJ"
        },
        {
            "nodes": [],
            "name": "Klever Stra\u00DFe",
            "location": [
                13.391237,
                52.552769
            ],
            "distance": 29.892836,
            "hint": "A1T-kgtU_pIsAAAAcgAAABMAAAAAAAAAVnPsQWgvmELP6U1BAAAAACwAAAByAAAAEwAAAAAAAADXFgEAhVXMAEHkIQNwVMwAcOMhAwEATxN2gVsJ"
        }
    ],
    "destinations": [
        {
            "nodes": [],
            "name": "Helmholtzstra\u00DFe",
            "location": [
                13.325698,
                52.521718
            ],
            "distance": 7.872453,
            "hint": "dpH_koaR_5JHAAAAEQAAAD4AAAAAAAAApjmfQr6mlkFS5odCAAAAAEcAAAARAAAAPgAAAAAAAADXFgEAglXLAPZqIQOiVcsAsmohAwMA7w92gVsJ"
        },
        {
            "nodes": [],
            "name": "Torstra\u00DFe",
            "location": [
                13.403316,
                52.529723
            ],
            "distance": 14.802468,
            "hint": "-b70kv___38FAAAAEgAAALAAAACcAgAA0wVvQJBUAkEyo-pCkWzSQwUAAAASAAAAsAAAAJwCAADXFgEAtITMADuKIQO4hMwAtokhAwQAjwV2gVsJ"
        }
    ],
    "fallback_speed_cells": [],
    "code": "Ok",
    "message": null
}